src/main/java/io/kestra/plugin/aws/kinesis/PutRecords.java

```diff
@@ -0,0 +1,295 @@
+
+    @PluginProperty(dynamic = true)
+    @Schema(
+        title = "List of record to put or internal storage uri to retrieve it.",
+        description = "A list of at least one record."
+    )
+    @NotNull
+    private Object records;
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        final long start = System.nanoTime();
+
+        List<Record> records = getRecordList(this.records, runContext);
+
+        PutRecordsResponse putRecordsResponse = putRecords(runContext, records);
+
+        // Set metrics
+        runContext.metric(Timer.of("duration", Duration.ofNanos(System.nanoTime() - start)));
+        runContext.metric(Counter.of("failedRecordCount", putRecordsResponse.failedRecordCount()));
+        runContext.metric(Counter.of("successfulRecordCount", records.size() - putRecordsResponse.failedRecordCount()));
+        runContext.metric(Counter.of("recordCount", records.size()));
+
+        // Fail if failOnUnsuccessfulRecords
+        if (failOnUnsuccessfulRecords && putRecordsResponse.failedRecordCount() > 0) {
+            var logger = runContext.logger();
+            logger.error("Response show {} record failed: {}", putRecordsResponse.failedRecordCount(), putRecordsResponse);
+            throw new RuntimeException(String.format("Response show %d record failed: %s", putRecordsResponse.failedRecordCount(), putRecordsResponse));
+        }
+
+        File tempFile = writeOutputFile(runContext, putRecordsResponse, records);
+        return Output.builder()
+            .uri(runContext.putTempFile(tempFile))
+            .failedRecordsCount(putRecordsResponse.failedRecordCount())
+            .recordCount(records.size())
+            .build();
+    }
+
+    private PutRecordsResponse putRecords(RunContext runContext, List<Record> records) throws IllegalVariableEvaluationException {
+        try (KinesisClient client = client(runContext)) {
+            PutRecordsRequest.Builder builder = PutRecordsRequest.builder();
+
+            if (!Strings.isNullOrEmpty(streamArn)) {
+                builder.streamARN(streamArn);
+            } else if (!Strings.isNullOrEmpty(streamName)) {
+                builder.streamName(streamName);
+            } else {
+                throw new IllegalArgumentException("Either streamName or streamArn has to be set.");
+            }
+
+
+            List<PutRecordsRequestEntry> putRecordsRequestEntryList = records.stream()
+                .map(throwFunction(record -> record.toPutRecordsRequestEntry(runContext)))
+                .collect(Collectors.toList());
+
+
+            builder.records(putRecordsRequestEntryList);
+            return client.putRecords(builder.build());
+        }
+    }
+
+    private List<Record> getRecordList(Object records, RunContext runContext) throws IllegalVariableEvaluationException, URISyntaxException, IOException {
+        if (records instanceof String) {
+            // Try to deserialize json string
```

Comment:
We never just blindly try and fallback.
It should be either a String or a list of object (or an Object if you want to support single object), and the deserialization layer should already take care of it.

So first check if it's a String, if so read it as an internal storage.
Then check if it's a list, then it will be a List<Map<String, Objecy>> as an object will be a Map<String, Object>.
Last, check if it's a Map, then it will be a single object.
If none of the above, throws.
You can check how it's done on AWS SQS for ex.

Reply:
I used that to support either a string to internal directory or a json encoded list. Do you think I should drop support for the json encoded string ?